### PR TITLE
Delete session before recovery

### PR DESCRIPTION
**Pull request by nsklikas**
IAM-1044

This is hacky, but I don't think there is a better way to implement it.

Fixes https://github.com/canonical/kratos-operator/issues/259

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,6 +53,7 @@ At the moment the application is sourcing the following from the environment:
   `log.txt`. **Make sure application user has permissions to write**.
 - `PORT` - HTTP server port, defaults to `8080`
 - `BASE_URL` - the base url that the application will be running on
+- `KRATOS_SESSION_COOKIE_NAME` - the kratos session cookie name, defaults to `ory_kratos_session`
```
Comment (shipperizer) [inline README.md:56]: i'd say let's make it a `const` and job on, it wont be changeable or at least, we won't be changing this

File: pkg/kratos/handlers.go [hunk 2]
```diff
@@ -560,12 +567,29 @@ func (a *API) handleCreateSettingsFlow(w http.ResponseWriter, r *http.Request) {
...
 }
 
-func NewAPI(service ServiceInterface, mfaEnabled bool, baseURL string, logger logging.LoggerInterface) *API {
+func (a *API) deleteKratosSession(w http.ResponseWriter) {
+	// To delete the session we delete the kratos session cookie.
+	// This is hacky as it does not call the Kratos API and is likely to break on
+	// a new Kratos version, but there is no easy way to delete the session
+	// from the Kratos API
+	c := &http.Cookie{
+		Name:     a.kratosSessionCookieName,
```
Comment (shipperizer) [inline pkg/kratos/handlers.go:576]:
```
const KRATOS_COOKIE_NAME = "whatever"
```

picking up on on the point above

```suggestion
		Name:     KRATOS_COOKIE_NAME,
```

**Review by pik4ez-canonical — DISMISSED** [review]

**Comment by pik4ez-canonical** [comment]
For me looks like acceptable level of hackiness at this stage.

**Comment by natalian98** [comment]
While this solves the issue from https://github.com/canonical/kratos-operator/issues/259, it breaks another case:
If you log in using password, then without inserting the totp code you go to `/ui/reset_password`, that error appears:
image

File: pkg/kratos/handlers.go [hunk 3]
```diff
@@ -580,6 +603,18 @@ func NewAPI(service ServiceInterface, mfaEnabled bool, baseURL string, logger lo
 	return a
 }
 
+func setCookiesWithExclude(w http.ResponseWriter, cookies []*http.Cookie, exclude []string) {
```
Comment (shipperizer) [inline pkg/kratos/handlers.go:606]:
```suggestion
func setCookies(w http.ResponseWriter, cookies []*http.Cookie, exclude ...string) {
```

I'd say to override setCookies with a variadic so we keep it the same and should not break backwards compatibility

Comment (nsklikas) [reply pkg/kratos/handlers.go]: ahh thanks. I knew there had to be a better way to do this, but I suck at go....

**Review by shipperizer — APPROVED** [review]

**Pull request merged**